src/meta/src/hummock/compaction/picker/compaction_task_validator.rs

```diff
@@ -89,19 +89,25 @@ struct TierCompactionTaskValidationRule {
 
 impl CompactionTaskValidationRule for TierCompactionTaskValidationRule {
     fn validate(&self, input: &CompactionInput, stats: &mut LocalPickerStatistic) -> bool {
+        // Limit sstable file count to avoid using too much memory.
+        let overlapping_max_compact_file_numer = std::cmp::min(
+            self.config.level0_max_compact_file_number,
+            MAX_COMPACT_LEVEL_COUNT as u64,
+        );
+
+        if input.total_file_count >= overlapping_max_compact_file_numer
```

Comment:
This makes subsequent line `input.total_file_count <= overlapping_max_compact_file_numer` useless (always true)

Reply:
ok, we can replace `input.total_file_count <= overlapping_max_compact_file_numer` with `true`.

I'm not quite sure that if we can directly receive the task that `total_file_count` > MAX_COMPACT_LEVEL_COUNT .How did you consider this ?

Reply:
Remove this redundant code and rest LGTM.